indico/config/config.py

```diff
@@ -0,0 +1,52 @@
+import os
+from pathlib import Path
+from indico.errors import IndicoInvalidConfigSetting
+from requests import Session
+
+
+class IndicoConfig:
+    """
+    Configuration for indico client 
+
+    Support setting configuration options either using environment variables or directly as keywords argument of this class
+    """
+    host: str 
+    protocol: str
+    serializer: str
+    api_token_path: str
+    api_token: str = None
+
+    def __init__(self, **kwargs):
+        self.host: str = os.getenv("INDICO_HOST", "app.indico.io")
+        self.protocol: str = os.getenv("INDICO_PROTOCOL", "https")
+        self.serializer: str = os.getenv("INDICO_SERIALIZER", "msgpack")
```

Comment:
Why is this default set to msgpack, wouldn't most response be from fog in the form of json? IndicoClient is instantiated with IndicoConfig but it doesn't appear to be using the serializer attribute of the config.